services/frontend-service/src/ui/utils/DexAuthProvider.tsx

```diff
@@ -0,0 +1,66 @@
+/*This file is part of kuberpult.
+
+Kuberpult is free software: you can redistribute it and/or modify
+it under the terms of the Expat(MIT) License as published by
+the Free Software Foundation.
+
+Kuberpult is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+MIT License for more details.
+
+You should have received a copy of the MIT License
+along with kuberpult. If not, see <https://directory.fsf.org/wiki/License:Expat>.
+
+Copyright freiheit.com*/
+import { jwtDecode } from 'jwt-decode';
+//import Cookies from 'js-cookie';
+import { TopAppBar } from '../components/TopAppBar/TopAppBar';
+import { Button } from '../components/button';
+import React from 'react';
+
+export const LoginPage: React.FC = () => {
+    // Define the navigation function using useCallback
+    const handleRedirect = React.useCallback(() => {
+        // Hard reload to invalidate cache and make the Dex call work.
+        window.location.reload();
+        window.location.href = '/login';
```

Comment:
I think if you first do a `reload` then it's not guaranteed that the next line is even executed.
```suggestion
        window.location.href = '/login';
```